src/org/thoughtcrime/securesms/PassphraseRequiredActionBarActivity.java

```diff
@@ -38,4 +44,9 @@ protected final void onCreate(Bundle savedInstanceState) {
   }
 
   protected void onCreate(Bundle savedInstanceState, boolean ready) {}
+
+  // "Locked Mode" is when the account is not configured (Welcome screen) or when sharing a backup
+  // (Add second device) and the user should not leave that screen otherwise the state would be lost.
+  // Sometimes it is fine to allow to change in these situations, like to see the logs or offline help.
```

Comment:
```suggestion
  // "Locked Mode" is when the account is not configured (Welcome screen)
  // or when sharing a backup (Add second device).
  // The app is "locked" to an activity if you will.
  // In "Locked Mode" the user should not leave that activity otherwise the state would be lost -
  // so eg. tapping app icon or notifications MUST NOT replace activity stack.
  // However, sometimes it is fine to allow to pushing activities in these situations, 
  // like to see the logs or offline help.
```

Reply:
> what's missing and a bit vague, also to me, is how "Locked Mode" is entered 

yes, both welcome activity and backup transfer activity extend from BaseActionBarActivity, so when merging they would set the "allow in locked mode" to true which is more clearer than the currently implicit way via the parent class